indexing-service/src/main/java/org/apache/druid/indexing/overlord/IndexerMetadataStorageAdapter.java

```diff
@@ -46,23 +47,31 @@ public IndexerMetadataStorageAdapter(
 
   public int deletePendingSegments(String dataSource, Interval deleteInterval)
   {
-    // Check the given interval overlaps the interval(minCreatedDateOfActiveTasks, MAX)
-    final Optional<DateTime> minCreatedDateOfActiveTasks = taskStorageQueryAdapter
+    // Find the earliest active task created for the specified datasource; if one exists,
+    // check if its interval overlaps with the delete interval.
+    final Optional<TaskInfo<Task, TaskStatus>> earliestActiveTaskStatusInfo = taskStorageQueryAdapter
         .getActiveTaskInfo(dataSource)
         .stream()
-        .map(TaskInfo::getCreatedTime)
-        .min(Comparator.naturalOrder());
+        .min(Comparator.comparing(TaskInfo::getCreatedTime));
 
-    final Interval activeTaskInterval = new Interval(
-        minCreatedDateOfActiveTasks.orElse(DateTimes.MAX),
-        DateTimes.MAX
-    );
+    if (earliestActiveTaskStatusInfo.isPresent()) {
+      final TaskInfo<Task, TaskStatus> theEarliestActiveTaskInfo = earliestActiveTaskStatusInfo.get();
```

Comment:
Nit:
Rather than using names like `earliestActiveInfo` and `theEarliestActiveInfo`, you might as well just call the first one, `optionalEarliestActiveInfo`.

Personal opinion:
You could even try simpler names without losing out on any useful information.
e.g. `earliestActiveTaskOptional` and `earliestActiveTask` (the suffix `StatusInfo` doesn't seem very useful here).

Reply:
yep, good suggestion!